### PR TITLE
Add support for wifi AX

### PR DESCRIPTION
**Pull request by andyrichardson**
## About

Hi there 👋  First time contributing to this project!

The changes here are none-breaking and are in preparation for recent/upcoming additions to:
 -  `iwinfo` [(see here)](https://git.openwrt.org/?p=project/iwinfo.git;a=commit;h=50b64a63e3945693646d1b2eb4e59e14e35cedc3)
  - `hostapd` [(see here)](https://patchwork.ozlabs.org/project/openwrt/list/?series=229531)
  
Special shoutout to @dangowrt for [piecing these changes together](https://github.com/dangowrt/linksys-e8450-openwrt-installer/issues/7#issuecomment-820328864).

This probably isn't going to be an immediate merge and could use some experienced eyes. I'll drop some comments below with sources.

### CODE REVIEW

File: modules/luci-mod-network/htdocs/luci-static/resources/view/network/wireless.js [hunk 1]
```diff
@@ -351,6 +368,13 @@ var CBIWifiFrequencyValue = form.Value.extend({
 				],
 				'ac': [
 					'11a', '5 GHz', true
+				],
+				// 'ad': [
+				//      '?', '60 GHz', true
```
Comment (jow-) [inline modules/luci-mod-network/htdocs/luci-static/resources/view/network/wireless.js:373]: Upstream hostapd.conf would use `hw_mode=ad` to select 60GHz, so in OpenWrt uci it'd likely get exposed as `option hwmode 11ad`

File: modules/luci-mod-network/htdocs/luci-static/resources/view/network/wireless.js [hunk 2]
```diff
@@ -336,7 +338,22 @@ var CBIWifiFrequencyValue = form.Value.extend({
...
 					'VHT80', '80 MHz', htmodelist.VHT80,
+					'VHT80+80', '160 MHz (dual 80 MHz)', htmodelist['VHT80+80'],
 					'VHT160', '160 MHz', htmodelist.VHT160
+				],
+				// 'ad': [
+				// 	'VHT20', '20 MHz', htmodelist.VHT20,
+				//	'VHT40', '40 MHz', htmodelist.VHT40,
+				//	'VHT80', '80 MHz', htmodelist.VHT80,
+				//	'VHT80+80', '160 MHz (dual 80 MHz)', htmodelist['VHT80+80'],
+				//	'VHT160', '160 MHz', htmodelist.VHT160
```
Comment (andyrichardson) [inline modules/luci-mod-network/htdocs/luci-static/resources/view/network/wireless.js:349]: - AD uses VHT [source](https://www.rohde-schwarz.com/uk/applications/802.11ad-wlan-at-60-ghz-white-paper_230854-50625.html)

File: modules/luci-base/htdocs/luci-static/resources/network.js [hunk 3]
```diff
@@ -3266,7 +3266,9 @@ WifiDevice = baseclass.extend(/** @lends LuCI.network.WifiDevice.prototype */ {
 	 *  - `b` - Legacy 802.11b mode, 2.4 GHz, up to 11 Mbit/s
 	 *  - `g` - Legacy 802.11g mode, 2.4 GHz, up to 54 Mbit/s
 	 *  - `n` - IEEE 802.11n mode, 2.4 or 5 GHz, up to 600 Mbit/s
-	 *  - `ac` - IEEE 802.11ac mode, 5 GHz, up to 6770 Mbit/s
+	 *  - `ac` - IEEE 802.11ac mode, 5 GHz, up to 1300 Mbit/s
+	 *  - `ad` - IEEE 802.11ad mode, 60 GHz, up to 7 Gbit/s
+	 *  - `ax` - IEEE 802.11ax mode, 2.4 or 5GHz, up to 3.5 Gbit/s (on a single stream)
```
Comment (andyrichardson) [inline modules/luci-base/htdocs/luci-static/resources/network.js]: - `ac` wave 1 speed [source](https://en.wikipedia.org/wiki/IEEE_802.11ac-2013#Wave_1_vs._Wave_2)
 - `ad` speed [source](https://www.rohde-schwarz.com/uk/applications/802.11ad-wlan-at-60-ghz-white-paper_230854-50625.html)
  - `ax` speed [source](https://ieeexplore.ieee.org/document/8468986/metrics#sec3)

File: modules/luci-mod-network/htdocs/luci-static/resources/view/network/wireless.js [hunk 4]
```diff
@@ -336,7 +338,22 @@ var CBIWifiFrequencyValue = form.Value.extend({
 					'VHT20', '20 MHz', htmodelist.VHT20,
 					'VHT40', '40 MHz', htmodelist.VHT40,
 					'VHT80', '80 MHz', htmodelist.VHT80,
+					'VHT80+80', '160 MHz (dual 80 MHz)', htmodelist['VHT80+80'],
```
Comment (andyrichardson) [inline modules/luci-mod-network/htdocs/luci-static/resources/view/network/wireless.js:341]: `VHT80+80` looks to be something we could be getting back from iwinfo [(see here)](https://git.openwrt.org/?p=project/iwinfo.git;a=blob;f=iwinfo_lib.c;h=70b080cedfc07be6b001f7084ad63ffe69d92f3f;hb=refs/heads/master#l60).

File: modules/luci-mod-network/htdocs/luci-static/resources/view/network/wireless.js (hunk 2 above)
Comment (dangowrt) [reply modules/luci-mod-network/htdocs/luci-static/resources/view/network/wireless.js:349]: At least QCA/Wilocity wil6210 reports 11n-like rates afaik, @lynxis and @robimarko will know more about that.

Comment (robimarko) [reply modules/luci-mod-network/htdocs/luci-static/resources/view/network/wireless.js:349]: Unfortunately, I don't have the wil6210 based Mikrotik now.
If I remember I can check later what it returns.

Comment (lynxis) [reply modules/luci-mod-network/htdocs/luci-static/resources/view/network/wireless.js:349]: just to clarify, this is about the phyrate from iwinfo?

Comment (dangowrt) [reply modules/luci-mod-network/htdocs/luci-static/resources/view/network/wireless.js:349]: @lynxis It's about `htmode` as used in `/etc/config/wireless` and reported by `iwinfo`. Afaik only `HT20` is allowed on wil6210 and that translates into **2160 MHz** channel width, right?

Comment (dangowrt) [reply modules/luci-mod-network/htdocs/luci-static/resources/view/network/wireless.js:349]: source: https://github.com/openwrt/openwrt/pull/2417#issuecomment-617744558

File: modules/luci-mod-network/htdocs/luci-static/resources/view/network/wireless.js [hunk 5]
```diff
@@ -351,6 +368,13 @@ var CBIWifiFrequencyValue = form.Value.extend({
 				],
 				'ac': [
 					'11a', '5 GHz', true
+				],
+				// 'ad': [
+				//      '11ad', '60 GHz', true
+				// ],
+				'ax': [
+					'11g', '2.4 GHz', true
```
Comment (dangowrt) [inline modules/luci-mod-network/htdocs/luci-static/resources/view/network/wireless.js:376]: Looks like you are missing a comma (`,`) at the end of this line

**Comment by andyrichardson** [comment]
Someone else fixed this elsewhere - closing

**Pull request closed without merging**